From c9146a65ab51c9c87c30e0c1b1d651e0953992df Mon Sep 17 00:00:00 2001
From: David Anderson <dvander@google.com>
Date: Mon, 1 Jun 2026 21:07:23 -0700
Subject: [PATCH 1/2] ANDROID: Add support for building exfat tests.

Bug: 508645975
Test: manual test
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:2bf87d7f951961b022139fca985c0c8b5827ac94
Merged-In: Idae8fb83f31f2f091bd3ac55a308d71e2eed9d8f
Change-Id: Idae8fb83f31f2f091bd3ac55a308d71e2eed9d8f
---
 Android.bp      | 4 +++-
 fsck/Android.bp | 1 +
 lib/Android.bp  | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index a4801ae..75e14d0 100644
--- a/Android.bp
+++ b/Android.bp
@@ -33,6 +33,7 @@ license {
 
 cc_library_headers {
     name: "libexfatprogs-headers",
+    host_supported: true,
     export_include_dirs: [
         "include",
         "mkfs",
@@ -40,12 +41,13 @@ cc_library_headers {
         "tune",
         "label",
         "dump",
-	"exfat2img",
+        "exfat2img",
     ],
 }
 
 cc_defaults {
     name: "exfatprogs-defaults",
+    host_supported: true,
     header_libs: ["libexfatprogs-headers"],
     export_header_lib_headers: ["libexfatprogs-headers"],
     cflags: ["-D_FILE_OFFSET_BITS=64"],
diff --git a/fsck/Android.bp b/fsck/Android.bp
index 7f63799..15a85af 100644
--- a/fsck/Android.bp
+++ b/fsck/Android.bp
@@ -11,6 +11,7 @@ package {
 
 cc_binary {
     name: "fsck.exfat",
+    host_supported: true,
 
     srcs: [
         "fsck.c",
diff --git a/lib/Android.bp b/lib/Android.bp
index 0ca169c..790bb9a 100644
--- a/lib/Android.bp
+++ b/lib/Android.bp
@@ -11,6 +11,7 @@ package {
 
 cc_library_static {
     name: "libexfat",
+    host_supported: true,
 
     srcs: [
         "libexfat.c",
-- 
2.55.0

